soc/arm/silabs_exx32/Kconfig

```diff
@@ -8,6 +8,13 @@ config SOC_FAMILY_EXX32
 	select BUILD_OUTPUT_HEX
 
 if SOC_FAMILY_EXX32
+
+config SOC_FAMILY_EFM32
+	bool
+
+config SOC_FAMILY_EFR32
+	bool
```

Comment:
I'm wondering if these should be treated as family, or as a SoC series in Zephyr. The usual way of organizing SoC families is to have them in separate directories in the `soc/<arch>` dir. This is not the case for SiLabs chips.

If the only reason we want to make the distinction between EFR32 and EFM32 is determining if the given SoC has radio capabilities, then maybe it would be better to just reduce it to something like `SOC_GECKO_HAS_RADIO`?